src/net.cpp

```diff
@@ -2556,6 +2556,8 @@ bool CConnman::Start(CScheduler& scheduler, const Options& connOptions)
         m_anchors = ReadAnchors(gArgs.GetDataDirNet() / ANCHORS_DATABASE_FILENAME);
         if (m_anchors.size() > MAX_BLOCK_RELAY_ONLY_ANCHORS) {
             m_anchors.resize(MAX_BLOCK_RELAY_ONLY_ANCHORS);
+        } else if (m_anchors.size() == 0) {
+            DeleteAnchorsFile(gArgs.GetDataDirNet() / ANCHORS_DATABASE_FILENAME);
```

Comment:
Why this change?

Reply:
Because if there is aren't anchors we can delete the file at that point, no need to wait to try connect them.

Reply:
We talked about it and I just remembered I can't remove it because in https://github.com/bitcoin/bitcoin/pull/24034/files#diff-00021eed586a482abdb09d6cdada1d90115abe988a91421851960e26658bed02R2029 if `m_anchors` is empty the anchors.dat won't be deleted.